src/resmom/mom_hook_func.c

```diff
@@ -3424,7 +3424,32 @@ void reply_hook_bg(job *pjob)
 	struct	batch_request *preq = pjob->ji_preq;
 #endif
 
-	if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_HERE) { /*MS*/
+	if (pjob->ji_hook_running_bg_on == IS_DISCARD_JOB) {
```

Comment:
I do not understand the purpose of ji_hook_running_bg_on. It stores inter-server/inter-mom/batch-request/batch-request+errno in it. These are all independent things, do you know why this one variable is used to store all of these things?

Reply:
ji_hook_running_bg_on is used indicate the hook is running in the background and it stores inter-server/inter-mom/batch-request/batch-request+errno in it to send appropriate reply from the deferred function (reply_bg_hook).

Reply:
This seems incorrect. A variable should not store 2 or more distinct things in it. Like in the case statement you check for "PBS_BATCH_DeleteJob + PBSE_SISCOMM" this will be same as "PBS_BATCH_Commit + PBSE_SVRDOWN". While the example above may not be practically possible but it is a possibility that at some point someone might hit this issue and it then it will be very difficult to diagnose.

Reply:
I see even it is two distinct values, will be unique for the event so there is little chance of getting into the issue.
The reason for holding these values is, when MS receives PBS_BATCH_DeleteJob request, it starts execution of the hook in the background and at the same time it sends the IM_DELETE_JOB_REPLY to all its sisters, if it is unable to send this request to one or more sisters, we need to reply the batch request with PBSE_SISCOMM error and this reply is done in the deferred function of hook running in the background, to carry this error value from function req_deletejob() to reply_hook_bg() we are using the ji_hook_running_bg_on variable.

Reply:
I hear what you are saying, but I don't like the fact that there is already a clash of values stored in this variable which is causing you to change the inter-mom request command IDs. This means that older moms and newer moms will not understand the messaged being passed between them (although, this may not be supported but it is possible, especially during upgrades)
I'd suggest that instead of storing existing batch request IDs, IM/IS command IDs, error number (or combinations of all three) you create a separate enum and add all possible values this variable is currently used for in that enum and then you don't have to change the Inter-mom command.